runhouse/resources/hardware/cluster.py

```diff
@@ -703,10 +717,12 @@ def restart_server(
         logger.info(f"Restarting Runhouse API server on {self.name}.")
 
         if resync_rh:
-            self._sync_runhouse_to_cluster(_install_url=_rh_install_url)
+            if not self.address:
+                raise ValueError(f"No address set for cluster <{self.name}>. Is it up?")
 
-        # Update the cluster config on the cluster
-        self.save_config_to_cluster()
+            for node in self.ips:
```

Comment:
For now leaving it like this until it can be unblocked - so far have tried: 
- `ProcessPoolExecutor`: Not running properly in a multiprocessing context
- `ThreadPoolExecutor`: Not running properly in multi-threaded approach, since the Live display in Rich is not designed to be used concurrently from multiple threads: 
```
        with self._lock:
            if self._live is not None:
>               raise errors.LiveError("Only one live display may be active at once")
E               rich.errors.LiveError: Only one live display may be active at once
```

Reply:
Have you tried combining `ProcessPoolExecutor` with an async event loop?

Something along the lines of:

```
        loop = asyncio.get_event_loop()
        result = await loop.run_in_executor(
            processPoolExecutor, ...
        )
```